src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1247,25 +1256,38 @@ class GrepOptions : public boost::noncopyable
 
    // derived from excludeFilePatterns
    std::vector<std::string> excludeArgs_;
-   bool gitFlag_;
-
+   
    void processExcludeFilePatterns()
    {
-      gitFlag_ = false;
       for (json::Value filePattern : excludeFilePatterns_)
       {
          if (filePattern.getType() != json::Type::STRING)
             LOG_DEBUG_MESSAGE("Exclude files contain non-string value");
          else
          {
             std::string excludeText = boost::algorithm::trim_copy(filePattern.getString());
-            if (excludeText.compare("gitExclusions") == 0)
-               gitFlag_ = true;
-            else if (!excludeText.empty())
-               excludeArgs_.push_back("--exclude=" + filePattern.getString());
+
+            if (gitFlag_)
+            {
+               excludeArgs_.push_back(":!" + excludeText);
+            }
+            else 
+            {
+               excludeArgs_.push_back("--exclude=" + excludeText);
+            }
          }
       }
-      excludeArgs_.push_back("--exclude-dir=cloud.noindex");
+
+      // always exclude cloud.noindex
```

Comment:
It'd be nice to have a way to set this via a user preference, in case users wanted to override (or append) to this list.

Reply:
This kind of was https://github.com/rstudio/rstudio/issues/10586 but perhaps this can be a user pref, or we just rely on the fact that `cloud.nonindex` can be selected by some `.gitignore` now that this option is more clearly expressed.

Reply:
As per https://github.com/rstudio/rstudio/issues/10529#issuecomment-1044608803 I went with removing the special casing.